### PR TITLE
Use dataclasses for Config & RelativeMutationID

### PR DESCRIPTION
**Pull request by EJahren**
I am adding type hints while trying to understand the code, and dataclasses made that a bit easier.

 The only changed behavior is that `repr(config)` is no longer `<__init__.Config object at ID>` and `hash(config)` is no longer `id(config)`.

### CODE REVIEW

File: mutmut/__init__.py [hunk 1]
```diff
@@ -802,10 +792,6 @@ def run_mutation(context: Context, callback) -> str:
         start = time()
         try:
             survived = tests_pass(config=config, callback=callback)
-            if survived and config.test_command != config._default_test_command and config.rerun_all:
-                # rerun the whole test suite to be sure the mutant can not be killed by other tests
-                config.test_command = config._default_test_command
-                survived = tests_pass(config=config, callback=callback)
```
Comment (boxed) [inline mutmut/__init__.py:808]: why is this removed?

**Comment by EJahren** [comment]
Hi @boxed! Sorry about the noise, I completely misunderstood how `_default_test_command` works, but I believe the current changes should be uncontroversial.

**Comment by boxed** [comment]
👍
